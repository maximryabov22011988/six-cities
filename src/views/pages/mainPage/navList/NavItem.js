import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const propTypes = {
  className: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  onChangeCity: PropTypes.func.isRequired
};

function NavItem({ className, city, onChangeCity }) {
  return (
    <li className="locations__item">
      <a
        className={classnames(className, 'locations__item-link')}
        href="#"
        onClick={onChangeCity(city)}
      >
        <span>{city}</span>
      </a>
    </li>
  );
}

NavItem.propTypes = propTypes;

export default NavItem;
