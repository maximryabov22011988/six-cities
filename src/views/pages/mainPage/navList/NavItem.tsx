import * as React from 'react';

import cn from 'classnames';

const propTypes = {
  city: PropTypes.shape({
    name: PropTypes.string,
    location: PropTypes.arrayOf(PropTypes.number),
    zoom: PropTypes.number,
  }).isRequired,
  className: PropTypes.string.isRequired,
  onChangeCity: PropTypes.func.isRequired,
};

function NavItem({ className, city, onChangeCity }) {
  return (
    <li className="locations__item">
      <a className={cn(className, 'locations__item-link')} href="#" onClick={onChangeCity(city)}>
        <span>{city.name}</span>
      </a>
    </li>
  );
}

NavItem.propTypes = propTypes;

export default NavItem;
