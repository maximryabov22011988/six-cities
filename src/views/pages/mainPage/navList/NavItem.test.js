import React from 'react';
import renderer from 'react-test-renderer';

import NavItem from './NavItem';

const cityMock = {
  name: 'Cologne',
  location: [50.938361, 6.959974],
  zoom: 13,
};

it('NavItem correctly render', () => {
  const handleChangeCity = jest.fn();
  const navItem = renderer
    .create(
      <NavItem
        className="tabs__item"
        city={cityMock}
        onChangeCity={handleChangeCity}
      />,
    )
    .toJSON();

  expect(navItem).toMatchSnapshot();
});
