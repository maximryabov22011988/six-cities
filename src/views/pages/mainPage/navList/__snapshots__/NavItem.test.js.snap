// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem correctly render 1`] = `
<li
  className="locations__item"
>
  <a
    className="tabs__item locations__item-link"
    href="#"
  >
    <span>
      Paris
    </span>
  </a>
</li>
`;
